File: 876. Middle of the Linked List.java
import java.util.Stack;

// * Definition for singly-linked list.
class ListNode {
     int val;
      ListNode next;
      ListNode() {}
      ListNode(int val) { this.val = val; }
      ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}

class Solution {

    Stack<ListNode> stack = new Stack<>();

    public ListNode middleNode(ListNode head) {

        ListNode cur = head;

        while(cur!=null){
            stack.add(cur);
            cur = cur.next;
        }

        int middle = stack.size()/2;

        while(stack.size()>middle+1){
            stack.pop();
        }


        return stack.peek();
    }
}

File: 876. Middle of the Linked List.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:

        count, cur = 0, head

        while cur != None:
            cur = cur.next
            count += 1
        mid = (count // 2) 
        while mid > 0:
            head = head.next
            mid -= 1

        return head

        